src/server.c

```diff
@@ -810,7 +814,7 @@ struct redisCommand redisCommandTable[] = {
      0,NULL,0,0,0,0,0,0},
 
     {"publish",publishCommand,3,
-     "pub-sub ok-loading ok-stale fast",
+     "pub-sub ok-loading ok-stale fast can-replicate",
```

Comment:
In theory, you're right. Since the flag is named "might", we can set it to EXEC and then write code to ignore it. 
But that logic can be applied to any other flag (if we try to put aside the "might" part). 
I. E. We can set the OOM flag and others. 

I'd like to offer a different way of thinking about EXEC: it's not a command, it's part of the redis client protocol. It would be wrong for anyone to look at its flags. Instead one must always looks the the flags of the commands inside it. 

Maybe it would have been better if it wasn't in the command table, but rather implemented like QUIT is. 

Anyway, I vote to keep the code s it is.

Reply:
I agree with Oran.